hoa2dot/core.py

```diff
@@ -370,6 +425,13 @@ def to_hoa_repr(self) -> str:
             s += "{" + " ".join(map(str, self.acc_sig)) + "}"
         return s
 
+    def __eq__(self, other):
+        return isinstance(other, State) and self.index == other.index and self.label == other.label and \
+            self.name == other.name and self.acc_sig == other.acc_sig
+
+    def __hash__(self):
+        return hash((self.index, self.label, self.name, self.name))  # being a set, acc_sig is unhashable
```

Comment:
worth considering `frozenset` type, the hashable version of `set`.

Reply:
yeah, that's a good idea! Do you want me to change it?

Reply:
sure!